tensornetwork/network_components.py

```diff
@@ -539,6 +539,10 @@ def __init__(self,
     if self.backend and not self.backend.dtype:
       self.backend.dtype = self._tensor.dtype
 
+    collection = ops.get_current_collection()
```

Comment:
I think it would be best to have this be in the init of `BaseNode` rather than `Node`

Reply:
done, moved it.